### PR TITLE
Initialize DADA and GUPPI header start times using either offset or start time

### PR DESCRIPTION
**Pull request by cczhu**
Allows setting of DADA and GUPPI header time and start time in their respective `fromvalues` methods by either passing `time` and `offset` or `time` and `start_time`.  (Once initialized, setting a mutable header's `time` will change its offset.)

Addresses #224.

### CODE REVIEW

File: baseband/guppi/header.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ def offset(self):
 
     @offset.setter
     def offset(self, offset):
-        self['PKTIDX'] = int(round((offset.to_value(u.s) / self['TBIN'] /
```
Comment (mhvk) [inline baseband/guppi/header.py]: Why this change? `to_value(u.s)` is not only allowed but quite a bit faster than first creating this value, turning it in a quantity, and then back into an array.

Comment (cczhu) [reply baseband/guppi/header.py]: I think (since I did it in the tests) that it's natural to try to pass a `TimeDelta` rather than a quantity to `offset`, which doesn't have the `to_value` attribute.

Comment (mhvk) [reply baseband/guppi/header.py]: OK, fair enough! That does remind me of https://github.com/astropy/astropy/issues/6148

**Comment by mhvk** [comment]
Would merge except that there is a pep 8 failure...
